experimental/unitsconversion/src/measureunit.rs

```diff
@@ -0,0 +1,271 @@
+    /// NOTE:
+    ///    if the prefix is found, the function will return (power, part without the prefix).
+    ///   if the prefix is not found, the function will return (0, part).
+    fn get_si_prefix_base_10(part: &str) -> (i8, &str) {
+        if let Some(part) = part.strip_prefix("quetta") {
+            (30, part)
+        } else if let Some(part) = part.strip_prefix("ronna") {
+            (27, part)
+        } else if let Some(part) = part.strip_prefix("yotta") {
+            (24, part)
+        } else if let Some(part) = part.strip_prefix("zetta") {
+            (21, part)
+        } else if let Some(part) = part.strip_prefix("exa") {
+            (18, part)
+        } else if let Some(part) = part.strip_prefix("peta") {
+            (15, part)
+        } else if let Some(part) = part.strip_prefix("tera") {
+            (12, part)
+        } else if let Some(part) = part.strip_prefix("giga") {
+            (9, part)
+        } else if let Some(part) = part.strip_prefix("mega") {
+            (6, part)
+        } else if let Some(part) = part.strip_prefix("kilo") {
+            (3, part)
+        } else if let Some(part) = part.strip_prefix("hecto") {
+            (2, part)
+        } else if let Some(part) = part.strip_prefix("deca") {
+            (1, part)
+        } else if let Some(part) = part.strip_prefix("deci") {
+            (-1, part)
+        } else if let Some(part) = part.strip_prefix("centi") {
+            (-2, part)
+        } else if let Some(part) = part.strip_prefix("milli") {
+            (-3, part)
+        } else if let Some(part) = part.strip_prefix("micro") {
+            (-6, part)
+        } else if let Some(part) = part.strip_prefix("nano") {
+            (-9, part)
+        } else if let Some(part) = part.strip_prefix("pico") {
+            (-12, part)
+        } else if let Some(part) = part.strip_prefix("femto") {
+            (-15, part)
+        } else if let Some(part) = part.strip_prefix("atto") {
+            (-18, part)
+        } else if let Some(part) = part.strip_prefix("zepto") {
+            (-21, part)
+        } else if let Some(part) = part.strip_prefix("yocto") {
+            (-24, part)
+        } else if let Some(part) = part.strip_prefix("ronto") {
+            (-27, part)
+        } else if let Some(part) = part.strip_prefix("quecto") {
+            (-30, part)
+        } else {
+            (0, part)
+        }
+    }
+
+    // TODO: consider returning Option<(i8, &str)> instead of (0, part) for the case when the prefix is not found.
+    // TODO: consider using a trie for the prefixes.
+    // TODO: complete all the cases for the prefixes.
+    /// Get the SI prefix for base 2.
+    /// NOTE:
+    ///     if the prefix is found, the function will return (power, part without the prefix).
+    ///     if the prefix is not found, the function will return (0, part).
+    fn get_si_prefix_base_two(part: &str) -> (i8, &str) {
+        if let Some(part) = part.strip_prefix("kibi") {
```

Comment:
nit: here you're searching from 0 to +inf, which I think makes sense. In the decimal units you do +inf -> 0 -> -inf. It probably makes sense to start at 0 and do +1 -> -1 -> +2 -> -2 ...

Reply:
I think it will be confusing:

"deca" , "deci" , "hecta", "centi", "kilo", "milli" .... etc.

Reply:
I don't think so, I see the pattern. Just a nit though

Reply:
ok